src/backend/internal/controller/user/update_user_profile.go

```diff
@@ -0,0 +1,104 @@
+package user
+
+import (
+	"errors"
+	"math/rand"
+	"net/http"
+
+	"github.com/Roongkun/software-eng-ii/internal/controller/user/fieldvalidate"
+	"github.com/Roongkun/software-eng-ii/internal/controller/util"
+	"github.com/Roongkun/software-eng-ii/internal/model"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func updateUserFieldsFromInput(userObj model.User, updatingUserInput model.UserUpdateInput) (model.User, error) {
```

Comment:
You could use `pass-by-reference` here to reduce the time used to copy the struct.

`updateUserField(userObj *model.User...)`